roles/confluent.common/tasks/remove_packages.yml

```diff
@@ -0,0 +1,44 @@
+---
+- name: Get Package Facts
+  package_facts:
+
+- set_fact:
+    confluent_package_version_changed: >
+      {{ ansible_facts.packages[main_package] is defined
+          and ansible_facts.packages[main_package][0].version
+            not in [confluent_package_version, confluent_full_package_version] }}
+
+- set_fact:
+    confluent_packages_removed: >
+      {{ ansible_facts.packages | dict2items | map(attribute='key') | select('match', match_pattern) | list }}
+  vars:
+    match_pattern: >
+      {{ 'confluent-' if confluent_package_version_changed
```

Comment:
should there be an else on this line?

Reply:
So its remove all packages if version changed or confluent-kafka-2.12 is defined? i was thinking only to remove all packages when the version changed.

Now im wondering what to do in the confluent-kafka-2.12 case.

Reply:
What if we just remove all confluent- packages in all remove cases? That will potentially remove "too much", but you should not get a remove case often, as it must be idempotent.

Reply:
Or use some set logic? Give me an hour or two! 😊

Reply:
One idea i have is: 
```
'confluent-kafka$' if ansible_facts.packages['confluent-kafka'] is defined and 'confluent-kafka' not in package_list
```
Where we pass the package list because what if you override that var
And repeat that for confluent-kafka/confluent-server/confluent-kafka-2.12